### PR TITLE
maintainers: update GitHub names

### PR DESCRIPTION
**Pull request by FliegendeWurst**
It is important to have a usable mapping of maintainers -> GH handles to ping maintainers for reviews or bugs.

This PR updates all outdated GitHub names based on the provided ID (note: I only fixed the casing up to JanLikar). I have included the checking script.

For the affected entries, I checked that the ID is correct by looking at one of their PRs.
- nerdypepper https://github.com/NixOS/nixpkgs/pull/134110/files
- osener https://github.com/NixOS/nixpkgs/pull/5838/files
- oxapentane https://github.com/NixOS/nixpkgs/pull/195000/files
- paddygord https://github.com/NixOS/nixpkgs/pull/48214/files
- patricksjohnson https://github.com/NixOS/nixpkgs/pull/192643/files
- pawelpacana https://github.com/NixOS/nixpkgs/pull/14980
- realsnick https://github.com/NixOS/nixpkgs/pull/217725/files
- sanana https://github.com/NixOS/nixpkgs/pull/341087
- tomsiewert https://github.com/NixOS/nixpkgs/pull/179450/files
- voidIess https://github.com/NixOS/nixpkgs/pull/95422/files

The following maintainers were not found. Probably they should be removed.
- cyewashish (account deleted)
- galaxy (account deleted)
- hikari (account deleted https://github.com/NixOS/nixpkgs/pull/230512)
- infinitivewitch (account deleted?)
- liberatys (account deleted?)
- lw (account deleted?)
- mjanczyk (account deleted?)
- ne9z (account deleted https://github.com/NixOS/nixpkgs/pull/223712)
- snowflake (account deleted? https://github.com/NixOS/nixpkgs/commit/fb8efafdea75b)
- vector1dev (account deleted https://github.com/NixOS/nixpkgs/pull/235623)

## Things done

- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Atemu** [comment]
Could you just simply put the script into the tree instead? Nobody is going to find this in a commit message.

We could even turn this into a CI check at some point.

**Comment by FliegendeWurst** [comment]
Script added in-tree.

The trouble with a CI check is that it will faily randomly. Perhaps it could be run periodically like the staging merges, and notify someone when it fails.

**Comment by Atemu** [comment]
Separate commit please.

Ah, I think I misunderstood the problem then. If this happens outside of our control, then yes a CI check doesn't make sense. We'd need a bot to run this regularly instead but I don't think this happens frequently enough to justify setting one up.

**Comment by philiptaron** [comment]
Contrary to @Atemu in this PR but in accordance with @Atemu in https://github.com/NixOS/nixpkgs/issues/327064, I'd prefer this _not_ go into the nixpkgs tree. @FliegendeWurst, want to make a tiny @nix-community repo for this? Or see what the delta between https://github.com/nix-community/rfc55 and your script is?

**Comment by FliegendeWurst** [comment]
> Or see what the delta between [nix-community/rfc55](https://github.com/nix-community/rfc55) and your script is?

That script seems to serve a different purpose. It looks through the committers team and finds inactive users.
My script checks the `maintainer-list.nix` for changed GH usernames.

**Review by philiptaron — APPROVED** [review]
I'm signing off on the updates to `maintainers/maintainer-list.nix`

File: maintainers/scripts/check-maintainer-usernames/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    real_main().await.unwrap();
+}
+
+async fn real_main() -> Result<(), Box<dyn Error>> {
+    let gh = octocrab::OctocrabBuilder::default()
+        .personal_token(env::var("GITHUB_PAT").expect("no GITHUB_PAT configured"))
+        .build()?;
+
+    // first run: nix-instantiate -A lib.maintainers --eval --strict --json > /tmp/x.json
+    let data: serde_json::Value = serde_json::from_str(&fs::read_to_string("/tmp/x.json").await?)?;
```
Comment (SuperSandro2000) [inline maintainers/scripts/check-maintainer-usernames/src/main.rs:16]:
```suggestion
    // first run: nix-instantiate -A lib.maintainers --eval --strict --json > maintainers.json
    let data: serde_json::Value = serde_json::from_str(&fs::read_to_string("maintainers.json").await?)?;
```

**Comment by SuperSandro2000** [comment]
> We could just skip the lock file. 

Yeah exactly. Having a repo for a 42 lines script is just creating repo brawl.

**Comment by FliegendeWurst** [comment]
> > We could just skip the lock file.
> 
> Yeah exactly. Having a repo for a 42 lines script is just creating repo brawl.

Removed as suggested. And it probably also makes more sense to keep this script next to all the other scripts in maintainers/scripts.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by Atemu — APPROVED** [review]
LGTM now.

I can't verify the script but it looks like it'd work and even if a slight error has crept in, someone could easily make it work.

**Comment by Atemu** [comment]
Ah, one last thing: Could you add the description you put into the PR message into the commit message too?

PRs can suddenly become non-permanent quickly and this would make it more accessible if the history is queried from git too.

I'd also order the commits the other way around but that's a nit.

**Comment by FliegendeWurst** [comment]
Description updated.
GitHub is stuck at ["Processing updates"](https://gh.io/pr-sync-in-progress), so I have opened #362419 instead.

**Pull request closed without merging**